### PR TITLE
Mini Reddit

### PR DESCRIPTION
**Pull request by SevlaMare**
-Setup
-Generate models
-Create migration files
-Set associations
-Set validations
-Write readme
-Linter fix

### CODE REVIEW

**Review by majovanilla — CHANGES_REQUESTED** [review]
Hey team!!

## Great job on this project! 👍

You made almost everything perfect!! Just a few things before moving to the next project.
- Please keep in mind that the model names matter on this side of the review. In this case, you changed the `user` model to `author` model and that's ok, but it would make our TSE's lives easier if you follow the instructions and name conventions on the Odin Project on the following projects!! :pray: 


There are still some changes 👇 you need to implement to approve this request 🔍 Please check the comments below and make the **required** changes.

If you want to make your project even better 🔥💥, I left a few suggestions you can implement.  Those are marked as **optional**.


If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Post < ApplicationRecord
+  belongs_to :author
+  has_many :comments, dependent: :destroy
+
+  validates :title, presence: true, uniqueness: true
```
[the remark this answers is not in the document]
Comment (SevlaMare) [reply app/models/post.rb:5]: We are thinking about having article with same title, can be bad when you try to search a specific article, but like you say, out there they allow duplicate titles.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  belongs_to :author
+  belongs_to :post
```
[the remark this answers is not in the document]
Comment (SevlaMare) [reply app/models/comment.rb:3]: We solve the required, but we are unable to solve the direct access to the object, like
`Post.first.comments.create(body: "some text")`

[the remark this answers is not in the document]
Comment (SevlaMare) [reply app/models/comment.rb:3]: We solve the required, but we are unable to solve, for now, the direct access to the object, like
`Post.first.comments.create(body: "some text")`

[the remark this answers is not in the document]
Comment (SevlaMare) [reply app/models/comment.rb:3]: We solve the required, but we are unable to set direct access to the object, like
`Post.first.comments.create(body: "some text")`

[the remark this answers is not in the document]
Comment (SevlaMare) [reply app/models/comment.rb:3]: We solve the required, but for now, we are unable to set direct access to the object, like
`Post.first.comments.create(body: "some text")`

**Review by jofer86 — APPROVED** [review]
## :wave: :wave: Hello!, I'm Jorge, and  I'm here to review this Pull Request, let's take a look shall we :eye: :eye: :eye: 
- Changes marked as **[OPTIONAL]** are not hard requirements but are nice to have.
- The required changes were met, this project is now complete!! 

# `status: approved` :rocket: :rocket: :+1: :+1: 
--------------------------------------------------------------------
Cheers, 
Jorge Fernando Rincón Figueroa.
Technical Support Engineer.
**Slack: @jofer86**

**Pull request merged**